src/coreclr/vm/instancecalli.cpp

```diff
@@ -0,0 +1,97 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "common.h"
+
+void MethodDesc::GenerateFunctionPointerCall(DynamicResolver** resolver, COR_ILMETHOD_DECODER** methodILDecoder)
+{
+    STANDARD_VM_CONTRACT;
+    _ASSERTE(resolver != NULL);
+    _ASSERTE(methodILDecoder != NULL);
+    _ASSERTE(*resolver == NULL && *methodILDecoder == NULL);
+    _ASSERTE(IsIL());
+    _ASSERTE(GetRVA() == 0);
+
+    // Intrinsic must be a static method.
+    _ASSERTE(IsStatic());
+
+    MetaSig declarationSig(this);
+    UINT argCount = declarationSig.NumFixedArgs();
+
+    // Intrinsic must have at least the "this" argument and the IntPtr function pointer argument.
+    _ASSERTE(argCount >= 2);
+
+    SigTypeContext genericContext;
+    ILStubLinker sl(
+        this->GetModule(),
+        GetSignature(),
+        &genericContext,
+        this,
+        (ILStubLinkerFlags)ILSTUB_LINKER_FLAG_NONE);
+
+    ILCodeStream* pCode = sl.NewCodeStream(ILStubLinker::kDispatch);
+
+    // Move to the function pointer argument.
+    declarationSig.SkipArg(); // Skip "this".
+
+    CorElementType fnType = declarationSig.NextArg();
+    _ASSERTE(fnType == ELEMENT_TYPE_FNPTR);
+
+    SigPointer spToken = declarationSig.GetArgProps();
+    spToken.GetByte(NULL); 
+    declarationSig.SkipArg();
+    SigPointer sigPtrTokenEnd = declarationSig.GetArgProps();
+
+    // Copy the existing signature so we can add HAS_THIS and EXPLICIT_THIS.
+    SigBuilder sigBuilder;
+    spToken.CopySignature(GetModule(), &sigBuilder, &sigPtrTokenEnd, IMAGE_CEE_CS_CALLCONV_HASTHIS | IMAGE_CEE_CS_CALLCONV_EXPLICITTHIS);
+
+    // Create a token for the new signature.
+    uint32_t sigLen;
+    PCCOR_SIGNATURE pSig = (PCCOR_SIGNATURE)sigBuilder.GetSignature((DWORD*)&sigLen);
+    mdToken fcnPtrToken = pCode->GetSigToken(pSig, sigLen);
+
+    // Create a local and store the function pointer value in it.
```

Comment:
It should not be necessary to store the function pointer in a local. The actual function pointer types do not matter in IL. It is all just pointers.

Reply:
Sure - removed (I was following the C# IL for release)